File: OpenRobertaServer/src/test/resources/crossCompilerTests/_expected/common/targetLanguage/ev3lejosv1/controlFlowNestedLoops.java
package generated.main;

import de.fhg.iais.roberta.runtime.*;
import de.fhg.iais.roberta.runtime.ev3.*;

import de.fhg.iais.roberta.mode.general.*;
import de.fhg.iais.roberta.mode.action.*;
import de.fhg.iais.roberta.mode.sensor.*;
import de.fhg.iais.roberta.mode.action.ev3.*;
import de.fhg.iais.roberta.mode.sensor.ev3.*;

import de.fhg.iais.roberta.components.*;

import java.util.LinkedHashSet;
import java.util.HashMap;
import java.util.Set;
import java.util.Map;
import java.util.List;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;

import lejos.remote.nxt.NXTConnection;

public class NEPOprog {
    private static Configuration brickConfiguration;

    private Set<UsedSensor> usedSensors = new LinkedHashSet<UsedSensor>();
    private Hal hal = new Hal(brickConfiguration, usedSensors);

    public static void main(String[] args) {
        try {
             brickConfiguration = new EV3Configuration.Builder()
                .setWheelDiameter(5.6)
                .setTrackWidth(18.0)
                .build();

            new NEPOprog().run();
        } catch ( Exception e ) {
            Hal.displayExceptionWaitForKeyPress(e);
        }
    }

    float ___X = 0;

    public void run() throws Exception {
        // Control Flow Nested Loop --Start
        if ( true ) {
            while ( true ) {
                while ( ! (___X >= 20) ) {
                    for ( float ___i = 1; ___i< 11; ___i+= 1 ) {
                        for ( float ___k0 = 0; ___k0< 2; ___k0+= 1 ) {
                            if ( (___i % 2 == 0) ) {
                                continue;
                            }
                            ___X += 1;
                        }
                    }
                }
                break;
            }
        }
        if (!(20 == ___X)) {
            System.out.println("Assertion failed: " + "pos-1" + 20 + "EQ" + ___X);
        }
        for ( float ___j = 1; ___j< 4; ___j+= 3 ) {
            ___X += 1;
            if (!(21 == ___X)) {
                System.out.println("Assertion failed: " + " = X" + 21 + "EQ" + ___X);
            }
            for ( float ___k = 1; ___k< 5; ___k+= 3 ) {
                ___X += 1;
            }
        }
        if (!(23 == ___X)) {
            System.out.println("Assertion failed: " + "pos-2" + 23 + "EQ" + ___X);
        }
        if ( true ) {
            while ( true ) {
                if ( true ) {
                    while ( true ) {
                        if ( 23 == ___X ) {
                            if ( true ) {
                                while ( true ) {
                                    if ( (___X % 2 == 0) ) {
                                        continue;
                                    }
                                    ___X += 1;
                                    break;
                                }
                            }
                            if ( (___X % 2 == 0) ) {
                                break;
                            }
                            ___X += 1000;
                        }
                        break;
                    }
                }
                break;
            }
        }
        System.out.println(( ( 24 == ___X ) ? "Control Flow Nested Loops Test:success" : "ntrol Flow Nested Loops Test: FAIL" ));
        if ( true ) {
            loop10:
            while ( true ) {
                while ( true ) {
                    if ( true ) {
                        if (true) break loop10;
                        break;
                    }
                    if ( true ) {
                        break;
                    }
                    hal.waitFor(15);
                }
            }
        }
        // Control Flow Nested Loop -- End
    }

}

File: OpenRobertaServer/src/test/resources/crossCompilerTests/_expected/common/targetLanguage/microbitv2/controlFlowNestedLoops.py
import microbit
import random
import math


class BreakOutOfALoop(Exception): pass
class ContinueLoop(Exception): pass

timer1 = microbit.running_time()

___X = 0

def run():
    global timer1, ___X
    # Control Flow Nested Loop --Start
    while True:
        while not (___X >= 20):
            for ___i in range(int(1), int(11), int(1)):
                for ___k0 in range(int(0), int(2), int(1)):
                    if (___i % 2) == 0:
                        continue
                    ___X += 1
        break
    if not 20 == ___X:
        print("Assertion failed: ", "pos-1", 20, "EQ", ___X)
    for ___j in range(int(1), int(4), int(3)):
        ___X += 1
        if not 21 == ___X:
            print("Assertion failed: ", " = X", 21, "EQ", ___X)
        for ___k in range(int(1), int(5), int(3)):
            ___X += 1
    if not 23 == ___X:
        print("Assertion failed: ", "pos-2", 23, "EQ", ___X)
    while True:
        while True:
            if 23 == ___X:
                while True:
                    if (___X % 2) == 0:
                        continue
                    ___X += 1
                    break
                if (___X % 2) == 0:
                    break
                ___X += 1000
            break
        break
    print("Control Flow Nested Loops Test:success" if ( 24 == ___X ) else "ntrol Flow Nested Loops Test: FAIL")
    while True:
        try:
            while True:
                if True:
                    raise BreakOutOfALoop
                    break
                if True:
                    break
        except BreakOutOfALoop:
            break
        except ContinueLoop:
            continue
    # Control Flow Nested Loop -- End

def main():
    try:
        run()
    except Exception as e:
        raise

if __name__ == "__main__":
    main()